formula-android/src/main/java/com/instacart/formula/integration/FragmentFlowRenderView.kt

```diff
@@ -169,10 +173,15 @@ internal class FragmentFlowRenderView(
     }
 
     private fun recordBackstackChange() {
-        val newBackstackSize = activity.supportFragmentManager.backStackEntryCount
-        if (backstackEntries > newBackstackSize) {
-            backstackPopped = true
+        val backStackEntryCount = backStackEntries.size
+        val newBackStackEntryCount = activity.supportFragmentManager.backStackEntryCount
+        if (backStackEntryCount > newBackStackEntryCount) {
+            val poppedFragmentName = backStackEntries.removeAt(backStackEntryCount - 1).name
```

Comment:
I wonder if we can assume that only `1` entry can be removed at the time.

Reply:
I actually probably shouldn't assume that (with transactions and other things, I wouldn't be surprised if it's not always the case + seems like fragment is slightly bug prone :trollface:, so we better protect ourselves)